### ISSUE TITLE
`repeat.for` on `<template>` element does not work in IE11

### ISSUE THREAD

**Issue by nenadvicentic** [aurelia/templating-resources#354]
**I'm submitting a bug report**

* **Library Version:**
1.8.1

aurelia-templating-router: 1.3.2
aurelia-templating-resources: 1.7.1
aurelia-templating-binding: 1.4.3
aurelia-binding: 2.1.1

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
8.11.2

* **Yarn Version:**
1.7.0

* **Browser:**
IE 11

* **Language:**
all

**Current behavior:**
`repeat.for` attribute of `<template>` element does not work in IE.

**Expected/desired behavior:**
`repeat.for` attribute of `<template>` should work in IE like it does in other browser (Chrome, FF, Edge)

Example markup, extended app.html generated by `aurelia-cli`:

```
<template>
  <h1>${message}</h1>
  
  <table>
      <thead>
        <tr>
          <td>Id</td>
          <td>Name</td>
        </tr>
      </thead>
      <tbody>
        <template repeat.for="item of dto.items">
          <tr>
            <td>${item.id}</td>
            <td>${item.name}</td>
          </tr>
        </template>
      </tbody>
    </table>
</template>

```

Extended typescript version of `App` class:

```

export class App {
  public dto: any;
  message = 'Hello World!';

  public attached(){
    setTimeout(() => {
      this.dto = {
        items: [
          {id: 1, name: "Jon Doe"},
          {id: 2, name: "Max Musterman"}
        ]
      }
    }, 1000);
  }
}

```

**Comment by Alexander-Taran** [comment]
https://github.com/aurelia/html-template-element/issues/3

**Comment by nenadvicentic** [comment]
I read the thread and it's companion https://github.com/aurelia/templating/issues/565. But I did not see any solution.

At the end, no proposed workaround was implemented? Like repeat syntax in the form of HTML comment or mentioned `template-start`, `template-end` attributes?

**Comment by Alexander-Taran** [comment]
@nenadvicentic for your example markup you can safely move repeat to `tr` tag
if you have more rows you can move repeat onto `tbody` tag which is proposed in the thread I linked.

**Comment by nenadvicentic** [comment]
@Alexander-Taran It's more then one row in real-life application. Repeated`tbody` tags will screw up table styling a bit, but if there is nothing better, it's fair enough.

**Comment by bigopon** [comment]
@nenadvicentic Well, it's been long known as a limitation of IE html parser. There is a work around for that, by constructing the view manually and use static view:

```js
export class App {
  static $view = {
    dependencies: [/* use this instead of <require/> */]
    template: h('template',
      h('h1', null, '${message}'),
      h('table', null,
        h('thead', null,
          h('tr', null,
            h('td', null, 'Id'),
            h('td', null, 'Name')
          )
        ),
        h('tbody', null,
          h('template', { 'repeat.for': 'items of dto.items' },
            h('tr', null,
              h('td', null, '${item.id}'),
              h('td', null, '${item.name}')
            )
          )
        )
      )
    )
  }

  public dto: any;
  message = 'Hello World!';

  public attached(){
    setTimeout(() => {
      this.dto = {
        items: [
          {id: 1, name: "Jon Doe"},
          {id: 2, name: "Max Musterman"}
        ]
      }
    }, 1000);
  }
}

import { DOM } from 'aurelia-framework';

function h(tag, props, ...children) {
  let el = tag === 'template' ? DOM.createTemplateElement() : DOM.createElement(tag);
  if (props) {
    for (let prop in props) {
      let propValue = props[prop];
      // simplified
      if (prop in el) {
        el[prop] = propValue;
      } else {
        el.setAttribute(prop, propValue);
      }
    }
  }
  let containerEl = tag === 'template' ? el.content : el;
  for (let i = 0, ii = children.length; ii > i; ++i) {
    const child = children[i];
    if (child === null || child === undefined) {
      continue;
    }
    // This is ... sad, probably not entirely correct but good enough for this snippet
    const appendable = child instanceof Node || child instanceof Comment || child instanceof Text;
    containerEl.appendChild(appendable ? child : DOM.createTextNode(child.toString());
  }
  return el;
}
```

It should work for IE

**Comment by nenadvicentic** [comment]
@bigopon I tried to test your example, because it looks very interesting to me, but I could not make it work. Do you know maybe what has to be configured in the startup, so that application does not try to load `app.html`, which I removed?

**Comment by bigopon** [comment]
@nenadvicentic You will need to use the following:

```
"aurelia-templating-router": "^1.3.2",
"aurelia-templating": "^1.8.0",
"aurelia-templating-resources": "^1.7.1",
"aurelia-binding": "^2.0.0"
```

For more info, you can have a look at https://aurelia.io/blog/2018/06/24/aurelia-release-notes-june-2018

**Comment by nenadvicentic** [comment]
@bigopon Double-checked. I am using correct versions. Problem was `setRoot(...)`. I don't know how to configure it, so that application does not to try to load both `.js` and `.html` file.  It might not even be possible.

But, I moved logic from `App` class to the custom element `my-table` and got application to start correctly. Unfortunately, then I've got errors with read-only properties on HTMLElements. I think it good idea what you proposed, but to cover all edge cases would probably end up being pretty complicated code.

I will stick with workaround with multiple `tbody` elements repeated at the moment and try to mitigate styling issues.
